### PR TITLE
[DEVX-1890] Add cucumber runner

### PR DESCRIPTION
**Pull request by tianfeng92**
Mac: https://app.saucelabs.com/tests/3ab56dc9b7224f10964bb19b6bf50a96
Windows: https://app.saucelabs.com/tests/523cb187bdd5419b9eb00d3af3018969

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,18 +26,23 @@
   "dependencies": {
     "@babel/core": "^7.14.6",
     "@babel/plugin-proposal-class-properties": "^7.17.12",
+    "@cucumber/cucumber": "8.6.0",
```
Comment (alexplischke) [inline package.json:29]: ⚠️ I thought the user had to provide cucumber. See DEVX-1803. If you provide cucumber prepackaged, the user will never be able to add any additional cucumber related dependencies. Or did you find a workaround for this?

File: package.json [hunk 2]
```diff
@@ -27,17 +27,21 @@
     "@babel/core": "^7.14.6",
     "@babel/plugin-proposal-class-properties": "^7.17.12",
     "@playwright/test": "1.25.1",
+    "@saucelabs/cucumber-reporter": "^0.2.0",
```
Comment (alexplischke) [inline package.json:30]: See my comment for https://github.com/saucelabs/sauce-playwright-runner/pull/171#discussion_r986270060, does this work?

Comment (tianfeng92) [reply package.json:30]: This reporter works 🤔 I set the cucumber bin as this way.
```
const cucumberBin = path.join(process.cwd(), 'node_modules', '@cucumber', 'cucumber', 'bin', 'cucumber-js');
```

Comment (alexplischke) [reply package.json:30]: Doesn't work for me: https://app.saucelabs.com/tests/77ab349bb01e463c856170d21ba7581d

Comment (tianfeng92) [reply package.json]: fixed in https://github.com/saucelabs/sauce-cucumber-reporter/pull/4/files

File: package.json [hunk 3]
```diff
@@ -33,6 +33,7 @@
     "glob": "^7.1.6",
     "lodash": "4.17.21",
     "playwright": "1.25.1",
+    "query-string": "^7.1.1",
```
Comment (alexplischke) [inline package.json:36]: That one still needed?

**Review by alexplischke — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (tianfeng92) [reply package.json:36]: yup. still needed here

**Pull request merged**